modules/conversantBidAdapter.js

```diff
@@ -27,6 +27,118 @@ const BIDDER_CODE = 'conversant';
 export const storage = getStorageManager({gvlid: GVLID, bidderCode: BIDDER_CODE});
 const URL = 'https://web.hb.ad.cpe.dotomi.com/cvx/client/hb/ortb/25';
 
+function setGDPR(bidderRequest, request) {
+  // Add GDPR flag and consent string
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(request, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+
+    if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+      deepSetValue(request, 'regs.ext.gdpr', bidderRequest.gdprConsent.gdprApplies ? 1 : 0);
+    }
+  }
+
+  if (bidderRequest.uspConsent) {
+    deepSetValue(request, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+}
+
+function setSiteId(bidRequest, request) {
+  if (bidRequest.params.site_id) {
+    if (request.site) {
+      request.site.id = bidRequest.params.site_id;
+    }
+    if (request.app) {
+      request.app.id = bidRequest.params.site_id;
+    }
+  }
+}
+
+function setPubcid(bidRequest, request) {
+  // Add common id if available
+  const pubcid = getPubcid(bidRequest);
+  if (pubcid) {
+    deepSetValue(request, 'user.ext.fpc', pubcid);
+  }
+}
+
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  request: function (buildRequest, imps, bidderRequest, context) {
+    const request = buildRequest(imps, bidderRequest, context);
+    request.at = 1;
+    request.cur = 'USD';
+    if (context.bidRequests) {
+      const bidRequest = context.bidRequests[0];
+      setSiteId(bidRequest, request);
+      setPubcid(bidRequest, request);
+    }
+
+    if (bidderRequest) {
+      setGDPR(bidderRequest, request);
+    }
+
+    return request;
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    const data = {
+      secure: 1,
+      bidfloor: getBidFloor(bidRequest) || 0,
+      displaymanager: 'Prebid.js',
+      displaymanagerver: '$prebid.version$'
+    };
+    copyOptProperty(bidRequest.params.tag_id, data, 'tagid');
+    mergeDeep(imp, data, imp);
+    return imp;
+  },
+  bidResponse: function (buildBidResponse, bid, context) {
+    if (!bid.price) return;
+    if (!context.mediaType && context.bidRequest.mediaTypes) {
+      const [type] = Object.keys(context.bidRequest.mediaTypes);
+      if (Object.values(ORTB_MTYPES).includes(type)) {
+        context.mediaType = type;
+      }
+    }
+    const bidResponse = buildBidResponse(bid, context);
+    return bidResponse;
+  },
+  response(buildResponse, bidResponses, ortbResponse, context) {
```

Comment:
I don't really want this in interpretResponse because I want ortbConverter to make all the decisions on what is returned.  Any decisions I make here may need to be modified if they make any changes in the future.

Reply:
Ok, let me ask another way.  I thought all the overrides should be adding or modding ORTB structures.  What does `response` do?  By returning just the `bids` portion here, didn't it just break the ORTB structure?

Reply:
I get it now.  You're right, I'll take care of it.